cronjobs/add_authority_external_ref._automation.py

```diff
@@ -0,0 +1,183 @@
+                    f"current version on the web: {current_file_date_integer}")
+                return True
+
+        else:
+            # The config file is not exist, assume it needs to create gnd_wiki file
+            UpdateConfigFile(config_file, "0")
+            return True
+
+    print("The file on the web is the same or older")
+    return False
+
+
+def Main():
+    # 1. Check whether the date of file on the web is newer compare to the last date successful update
+    print("Process 1/7 -- Check whether the file on the web is newer")
+    if IsThereANewlyReleasedFile():
+        # a. Download the newer file
+        newer_gz_file_name = "authorities-gnd-person_lds_" + \
+            str(current_file_date_integer) + ".jsonld.gz"
+        newer_file_name = "authorities-gnd-person_lds_" + \
+            str(current_file_date_integer) + ".jsonld"
+        print("Process 2/7 -- Downloading file")
+        if DownloadTheFile(newer_gz_file_name):
+            # b. The download file is a zip file, it needs to be extracted first
+            print("Process 3/7 -- Extracting the new file (.gz)")
+            util.ExecOrDie(util.Which("gunzip"), [
+                           "-f", f"/tmp/{newer_gz_file_name}"])
+            #  c. Get the information needed by gnd_wiki and put it into the file
+            print(
+                "Process 4/7 -- Parse the file and extract the essential information needed")
+            jq_prog_with_pipe = f"jq -c --stream '.' < /tmp/{newer_file_name} | grep -E 'https\:/\/d-nb\.info\/gnd\/|wikidata|wikipedia' > /tmp/{input_file_name_for_add_auth}"
+            if os.system(jq_prog_with_pipe) == 0:
+                # d. Run "add_authority_external_ref" program
+                print("Process 5/7 -- Generate a new gnd_wiki_file")
+                util.ExecOrDie(util.Which("add_authority_external_ref"), [
+                               "--create_mapping_file", f"/tmp/{input_file_name_for_add_auth}", f"/tmp/{gnd_wiki_file}"])
+
+                print("Process 6/7 -- Copy the gnd_wiki_file to its destination")
+                util.ExecOrDie(util.Which("cp"), [
+                               "-f", f"/tmp/{gnd_wiki_file}", f"{share_folder}{gnd_wiki_file}"])
+
+                #  e. Update the latest version date on the config file (the second line)
+                print(
+                    "Process 7/7 -- Updating the config file and remove temporary files")
+                UpdateConfigFile(
+                    config_file, current_file_date_integer)
+                util.ExecOrDie(util.Which("rm"), [
+                               "-f", f"/tmp/{newer_file_name}"])
+                util.ExecOrDie(util.Which("rm"), [
+                               "-f", f"/tmp/{input_file_name_for_add_auth}"])
+                util.ExecOrDie(util.Which("rm"), [
+                               "-f", f"/tmp/{gnd_wiki_file}"])
+
+
+try:
+    Main()
+except Exception as e:
+    print("Checking last update of file is error: " +
```

Comment:
Since Main() contains all the logic (not only checking but also downloading), the error message in the except branch is not correct.

Reply:
Also: We should send some kind of email to the admins with either a success message or an error message. There is a function in util.py SendEmail that can be used for this purpose.